draft-ietf-cbor-7049bis.md

```diff
@@ -962,11 +970,17 @@ for some of these formats.
   as defined in {{RFC4648}};
 
 * Tag 35 is for regular expressions in Perl Compatible Regular
-  Expressions (PCRE) / JavaScript syntax {{ECMA262}}.
-
+  Expressions (PCRE) format. {{PCRE}}, non-normative.
+  
 * Tag 36 is for MIME messages (including all headers), as defined in
   {{RFC2045}};
 
+* Tag ***TBD*** is for ECMAScript 2017 RegExp (Regular Expression) objects, 
```

Comment:
What are some applications that encode regular expressions in CBOR objects and want to mark them with a tag? Do we really need the core specification to reserve two tags for them?

Reply:
While I can't cite any use of Regex in CBOR, I've dealt with them a lot in high-volume time sensitive applications, e.g. spam filtering, where CBOR might be used in the future.  Semantic compatibility between different flavors (even different version numbers) can be a showstopper of a problem.   

RFC 7049 shows little understanding of this when it lumps PCRE and JavaScript regex in the same bucket.   Simply annotating a text string with tag 35 doesn't tell anyone if it's safe to assume their platform's Regex implementation is compatible with the regex stored in CBOR.   

Yet I see your point, allocating a new tag for each flavor isn't justified.   I'll remove it for now, it is more complicated than I thought through originally.

What might worth thinking about a mechanism where there's a tag that says "the following item is an array, a "flavor" may be encoded into the first item in the array.   For regex, an array might includes two or three elements, the first of which describes the flavor of the regex, the second is the textual representation in that's flavor's regex language, and the third describes the 'mode' of the regex.   

A question in my mind is do we dare do that (re)using tag 35?   RFC 7049 isn't completely clear about usage, but page 14 says it applies only to a UTF-8 string major type.   So I would expect that any existing implementation that uses tag 35 would be broken if it were applied to an array, causing a backward compatibility problem.   Dang nab it!

Overall, I think that tag 35 was underspecified when RFC 7049 was written, and there may be some lessons about tags in general to learn from this and incorporate into BIS:   

* When a tag is defined on a well-defined and stable standard, for example tag 33 and 34 for Base64 encodings defined in RFC 4648 (a proposed standard), it may be appropriate to define its application to only a single major type.   However, the following might be applicable.

* When a tag is defined related to a less stable and standardized format, such as regexes, you definitely want to include an array escape mechanism such as the 'flavor' mechanism I just described.   

* Even more flexible would be a fallback that depends on a map with a distinguished key that acts as fallback value for application code expecting only a string but receives a map..    The YAML type system defined such distinguished fallback key with a tag they called [`!!value`](http://yaml.org/type/value.html).   CBOR could benefit from doing this either as a tag, or perhaps a simple value using major type 7.

Reply:
Earlier, in responding to a question from @jyasskin. I agreed:

> allocating a new tag for each flavor isn't justified

My agreement still stands, this is not a problem CBOR needs to solve with a proliferation of Regex tags.    The point of the CR was updating the ECMAScript standard reference, the latest version is ECMAScript-2018.   I believe I did remove the portion of the change which proposed a PCRE tag